pkg/ipam/poolallocator/allocator.go

```diff
@@ -181,6 +181,35 @@ func (a *IPPoolAllocator) updateIPAddressState(ipPool *v1alpha2.IPPool, ip net.I
 
 }
 
+func (a *IPPoolAllocator) appendPoolUsageForStatefulSet(ipPool *v1alpha2.IPPool, ips []net.IP, namespace, name string) error {
+	newPool := ipPool.DeepCopy()
+
+	for i, ip := range ips {
+		owner := v1alpha2.IPAddressOwner{
+			StatefulSet: &v1alpha2.StatefulSetOwner{
+				Namespace: namespace,
+				Name:      name,
+				Index:     i,
+			},
+		}
+		usageEntry := v1alpha2.IPAddressState{
+			IPAddress: ip.String(),
+			Phase:     v1alpha2.IPAddressPhasePreallocated,
+			Owner:     owner,
+		}
+
+		newPool.Status.IPAddresses = append(newPool.Status.IPAddresses, usageEntry)
+	}
+	_, err := a.crdClient.CrdV1alpha2().IPPools().UpdateStatus(context.TODO(), newPool, metav1.UpdateOptions{})
+	if err != nil {
+		klog.Warningf("IP Pool %s update with status %+v failed: %+v", newPool.Name, newPool.Status, err)
+		return err
+	}
+	klog.InfoS("IP Pool update succeeded", "pool", newPool.Name, "allocation", newPool.Status)
```

Comment:
It's too verbose to log the whole status of IPPool for each update in default level.

Reply:
Moved to V2